### PR TITLE
Provide fix for broken parcelable extras, and support for the old drozer format for extras.

### LINKED ISSUE #458: drozer no longer supports old intent URI scheme and broken parcelable intents
Previous versions of drozer supported a slightly modified version of the `intent://` URI scheme that did not require wrapping the extras in `Intent` and `end`, as can be seen in the documentation under `help intents`:

> ```
> For Intents, it uses a slightly modified version of the 'intent://' format of URI:
>
>   --extra parcelable yayintentyay
> intent://host.com?queryparameter=value#package=targetPackage;component=targetComponent;action=action;category=category;type=type;identifier=identifier;launchFlags=launchFlags;scheme=scheme;sourceBounds=sourceBounds;S.extraString=value;B.extraBoolean=value;b.extraByte=value;c.extraChar=value;d.extraDouble=value;i.extraInt=value;f.extraFloat=value;l.extraLong=value;s.extraShort=value;
> ```

However, in modern versions this is unsupported, and this change is not reflected in the documentation.

Additionally, the `content://`, `file://`, and `http(s)://` URI schemes for a parcelable extra are non-functional due to an undefined variable `yayUriClassyay`.

### PR DESCRIPTION
**Pull request by nyaalex**

### CODE REVIEW

**Comment by nyaalex** [comment]
n.b. I also removed the specific check for `launchFlags`, I did this because that should be supported natively by Android's URI parser (seen [here](https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/core/java/android/content/Intent.java;l=8184-8190)).

**Comment by Yogehi** [comment]
one issue

you took out the following code:

```
while yayIntyay != 0:
yayKeyyay = yayUriArryay[yayIntyay - 1].split("=")[0]
yayValueyay = yayUriArryay[yayIntyay - 1].split("=")[1]
if yayKeyyay == "launchFlags":
    yayParcelableIntentyay.addFlags(context.arg(int(yayValueyay), obj_type="int"))
yayIntyay = yayIntyay - 1
```

the issue is that android's internal intent parser doesn't add launch flags for some reason. so even if you specify a launch flag in your `intent://` URI, the flag won't get added to the Intent Parcelable Extra

so you need to have code which manually adds the launch flag

**Comment by nyaalex** [comment]
Cool cool, that should be reimplemented :+1:

**Comment by nyaalex** [comment]
Oh, I've just realised why, I didn't even read the Android source _I linked_ :sweat_smile:. To read the `launchFlags` from a URI you need the `URI_ALLOW_UNSAFE` flag enabled. I'll need to test that out quickly but it should be a neater solution than my hacky regex one or even the OG loop through the extras. (Edit: confirmed to be working on my Android 11 test device. Given this was a feature implemented in API version 22 it's fair to say this works)

File: src/drozer/console/session.py [hunk 1]
```diff
@@ -540,7 +540,7 @@ def help_intents(self):
 
         For Intents, use the 'intent://' format of URI:
 
-          --extra parcelable yayintentyay intent://host.com?queryparameter=value#Intent;package=targetPackage;component=targetComponent;action=action;category=category;type=type;identifier=identifier;launchFlags=launchFlags;scheme=scheme;sourceBounds=sourceBounds;S.extraString=value;B.extraBoolean=value;b.extraByte=value;c.extraChar=value;d.extraDouble=value;i.extraInt=value;f.extraFloat=value;l.extraLong=value;s.extraShort=value;end;
+          --extra parcelable yayintentyay intent://host.com?queryparameter=value#Intent;package=targetPackage;component=targetPackage/.targetComponent;action=action;category=category;type=type;identifier=identifier;launchFlags=launchFlags;scheme=scheme;sourceBounds=sourceBounds;S.extraString=value;B.extraBoolean=value;b.extraByte=value;c.extraChar=value;d.extraDouble=value;i.extraInt=value;f.extraFloat=value;l.extraLong=value;s.extraShort=value;end;
```
Comment (cyberMilosz) [inline src/drozer/console/session.py]: Realistically, we will need better documentation **with examples** somewhere else, but for now:

- I'd suggest we drop the mention of `package` here - you can achieve the same results with just `component` (I can't think of a case where specifying both makes sense)
- I'd write it down as `targetPackage/targetComponent` - I'd argue that (when it applies) the leading dot is part of the name of the target component. `com.somepackage/com.somepackage.somecomponent` or `com.somepackage/org.something.else.entirely` would both be perfectly valid components, too

**Pull request merged**